### PR TITLE
ODIM-589 Store metric data in db

### PR DESCRIPTION
**Pull request by irappa-pattar**
Get below metric data from southbound and store it in DB

- MetricDefinitionCollection
- MetricReportDefinitionCollection
- MetricReportCollection
- TriggerCollection

### CODE REVIEW

**Review by jeevan-kamkar — APPROVED** [review]

**Review by shivacharanms — APPROVED** [review]

File: svc-telemetry/telemetry/telemetry.go [hunk 1]
```diff
@@ -344,18 +292,10 @@ func (e *ExternalInterface) GetMetricDefinition(req *teleproto.TelemetryRequest)
...
-	resp.Header = map[string]string{
-		"Allow":             `"GET"`,
-		"Cache-Control":     "no-cache",
-		"Connection":        "keep-alive",
-		"Content-type":      "application/json; charset=utf-8",
-		"Transfer-Encoding": "chunked",
-		"OData-Version":     "4.0",
-	}
-
+	resp.Header = header
```
Comment (bharath-b-hpe) [inline svc-telemetry/telemetry/telemetry.go:295]: common header defined has only "GET" as value for "Allow", for triggers PATCH also should be included.

File: svc-telemetry/telemetry/telemetry.go [hunk 2]
```diff
@@ -375,15 +315,7 @@ func (e *ExternalInterface) GetTrigger(req *teleproto.TelemetryRequest) response
...
```
Comment (bharath-b-hpe) [inline svc-telemetry/telemetry/telemetry.go:318]: common header defined has only "GET" as value for "Allow", for triggers PATCH also should be included.

**Review by bharath-b-hpe — CHANGES_REQUESTED** [review]
Please resolve the changes reported by lgtm tool too

File: svc-telemetry/telemetry/telemetry.go (hunk 1 above)
Comment (irappa-pattar) [reply svc-telemetry/telemetry/telemetry.go:296]: addressed

File: svc-telemetry/telemetry/telemetry.go (hunk 2 above)
Comment (irappa-pattar) [reply svc-telemetry/telemetry/telemetry.go:320]: addressed

**Review by bharath-b-hpe — APPROVED** [review]

**Pull request merged**